### DISCUSSION TITLE
What Should I learn

### DISCUSSION THREAD

**Discussion by Brayden925** [codewars/codewars.com#3062] [General]
I am currently learning HTML and CSS and school but I want to learn a programming language aswell. What language, if any, would work best with what I already know?

**Comment by raslvsu** [comment]
I think you should probably start with javascript if your goal is web development

**Reply by mousewanderer** [reply]
recommended to JavaScript, then focusing on creating logic and the fundamentals. after that I suggest on building small projects

**Comment by shivaamm** [comment]
You may start with JavaScript if you want to continue with some logic building language while using it for your development purpose, otherwise I would suggest you to learn JAVA. 

I was on the same track but I was already having some knowledge of C++, but now I work in Java.
@Brayden925

**Comment by Berzerk04** [comment]
As with the other responses, JavaScript is a must-learn language for web development. It's used to add interactivity and dynamic behavior to websites. With JavaScript, you can create things like form validation, interactive web applications, and more. Since you're already learning HTML and CSS, adding JavaScript to your skill set will make you a more well-rounded web developer.

I also suggest learning Python. It is a versatile and beginner-friendly language and while it's not directly related to web development like JavaScript, it's widely used for server-side web development (with frameworks like Django and Flask) and for tasks like web scraping, data analysis, and automation. Learning Python can complement your web development skills.

**Comment by ahmadshah2103** [comment]
- In my opinion, to learn any programming language, you should first build your logic building skills.
- **What programming language is best to start with?** The answer to this question depends, if you have good logic building skills, go with a language that works in a feild you are intersted in (like _JavaScript_ in case you are interested in _Web Development_)
- **And**, if you don't have logic building skills, I would recommend **C++**.

**Reply by karenworld** [reply]
why did you say that if you don't have logic building skills, use c++ ? I thought c++ is considered a difficult language to learn as it doesn't have auto garbadge collection?

**Comment by SHah-Omid** [comment]
If your goal is to become a web developer then start learning JavaScript

**Comment by Dest-M** [comment]
Javascript is the best, but I suggest to poke around in other languages as well, at least to get to know basic syntax and algoritms. There are many websites you could use, but so far Metanit is one of the more in depth and learner friendly that I've seen.

Another fun utility language is batch script - not very useful in terms of getting a job, but fun to play around with and easy to learn. You can use it to make simple bots and automate tasks - just in the notepad app. I used to love it when I was at school.

Good luck on your programming journey!

**Comment by shrishti92** [comment]
I suggest learning JavaScript.

**Comment by SuitcaseCoder** [comment]
I may be biased as the founder of the company, but I built [Code Flight](https://www.codeflight.io/) that begins every user with a discovery quiz to help them identify what their ideal coding path is for them and then curates a curriculum based on those unique goals

**Comment by Prafic** [comment]
> I am currently learning HTML and CSS and school but I want to learn a programming language aswell. What language, if any, would work best with what I already know?

**Comment by Prafic** [comment]
1 st  java then if you have time learn C ++

**Comment by Fillare** [comment]
Dear Brayden , I suggest you to learn python because it's syntax is simple and similar to English language. It can be used in web-development, app-development , machine-learning , deep-learning , system-scripting and much more . When i didn't even know a word about programming languages , i chose python , that was one of the best decision i ever took.

**Comment by Peixe2b** [comment]
I recommend Javascript or typescript, I also recommend Python with the flask library for the backend, It's very easy, in 1-3 months you already get it but it depends on your familiarity

**Reply by Armanman33** [reply]
mate i have started learning pythong for almost 3 months and each day 2hours, ( iwasnt that deciplined tbh) but i still have problems in simple functions. am i in wrong path?

**Comment by juhaina1996** [comment]
Since you're already learning HTML and CSS, JavaScript is the best programming language to learn next because it smoothly integrates with them to build dynamic web sites. Understanding JavaScript will enable you to create dynamic websites and improve your front-end programming skills.

**Comment by devkhishan** [comment]
I would suggest learn Rust and start doing web assembly, along with that you could able to learn to write low level code and best speed on competitive programming and also can be used in web assembly.  Learn some big stuffs continuously because it would be more interesting - do only if you are enthusiastic and not worried about spending more time on learning.

**Comment by ogeobubu** [comment]
It's great to hear that you're already learning HTML and CSS! These are the cornerstones for front-end development. JavaScript would be a natural next step. It is the primary language used to create dynamic and interactive web applications.

**Comment by YeshanSheranga** [comment]
If you are interested in Web development and If you have a good knowledge for HTML and  CSS then move to the Javascript. Start the basics and go to advanced include Document object Model part. then start the small projects using HTML CSS and Javascript. those three are frontend technologies

**Comment by Edberaga** [comment]
It depends on which development you would like to do in the future. For example, each development has a programming language best to use:
- Web Development = JavaScript
- Machine Learning = Python
- Game Development = C#
- Mobile App Development = Java/Kotlin
- Desktop App Development = Go

But overall, all kind Development **JavaScript** can be used for all of it. so, if you also confused which Development you would like to go, you can always learn **JavaScript**.

It did be best you master one Programming language first and it's JavaScript because JavaScript is kind of everywhere. But to be the best programmer we don't need to master programming language just at least familiar with them 2 or 3. After JavaScript I would recommend Java/C#/Python. Most large companies use these 3, just between one of them.

**Reply by Armanman33** [reply]
can you explain please why did u said we dont need to master the language? what did u mean?

**Comment by jocypher** [comment]
> I am currently learning HTML and CSS and school but I want to learn a programming language aswell. What language, if any, would work best with what I already know?

I think since you are learning html and css you can add javascript and then focus on the area you would want to pursue. There are many diverse fields in development however javascript is a great start

**Comment by UmarTheCoder2** [comment]
guys where is python for people who doesnt know anything

**Reply by UmarTheCoder2** [reply]
.....

**Reply by UmarTheCoder2** [reply]
ok better do it myself

**Reply by zaidkhuroo** [reply]
I would suggest to buy this course and get started.  [https://www.udemy.com/course/learn-python-with-abdul-bari/?couponCode=MT24125](url)

**Comment by zaidkhuroo** [comment]
javascript is a great choice to start with,followed by DSA.

**Comment by Mehrouf12z** [comment]
hi i wanna be a full stack dev and can someoene help me with a proper road map for it
i know html and css  and im starting to learn js as well and also python
and ik react as well but i wanna master all this

**Comment by RanRosa90** [comment]
Quick questioon, what are effective of ways of understand what is being asked of me for the problem,  like phrases in word problems that trigger addition or subtraction , how do i know or figure out how to build the type of code im bbein asked .... im only i 8 kyu pythonm

**Comment by hobovsky** [comment]
This type of discussion is off-topic for GitHub. We handle social and learning matters on our [Discord server](https://discord.gg/mSwJWRvkHA) and its `#ask-help` and `#beginner-intros` channels, where it’s easier, and safer for everyone to join public channels and learn together.

It’s much more effective (and fun!) to share progress and ask questions in open channels with man users rather than private groups. Please continue this conversation on Discord instead.
